src/client.js

```diff
@@ -309,24 +288,38 @@ assign(Client.prototype, {
   // returning a promise resolved when the connection is released.
   releaseConnection(connection) {
     debug('releasing connection to pool: %s', connection.__knexUid)
-    return this.pool.release(connection).catch(() => {
+    const didRelease = this.pool.release(connection)
+
+    if (!didRelease) {
       debug('pool refused connection: %s', connection.__knexUid)
-    })
+    }
+
+    return Promise.resolve()
   },
 
   // Destroy the current connection pool for the client.
   destroy(callback) {
-    return Promise.resolve(
-      this.pool &&
-      this.pool.drain()
-        .then(() => this.pool.clear())
-        .then(() => {
-          this.pool = void 0
-          if(typeof callback === 'function') {
-            callback();
-          }
-        })
-    );
+    let promise = null
+
+    if (this.pool) {
+      promise = this.pool.destroy()
```

Comment:
`destroy` drains and clears the pool just like `drain` + `clear` with the old pool.